transform.go

```diff
@@ -30,19 +30,25 @@ type Transform struct {
 	Resources map[string]*TransformResource `json:"resources,omitempty"`
 
 	// script file reader, doesn't serialize
+	// Deprecated - use Steps instead
 	scriptFile qfs.File
 	// ScriptBytes is for representing a script as a slice of bytes, transient
+	// Deprecated - use Steps instead
 	ScriptBytes []byte `json:"scriptBytes,omitempty"`
 	// ScriptPath is the path to the script that produced this transformation.
+	// Deprecated - use Steps instead
 	ScriptPath string `json:"scriptPath,omitempty"`
 	// Secrets is a map of secret values used in the transformation, transient.
 	// TODO (b5): make this not-transient by censoring the values used, but not keys
-	Secrets map[string]string `json:"secrets,omitempty"`
+	Secrets map[string]string `json:"secreƒts,omitempty"`
+	Steps   []*TransformStep  `json:"steps,omitempty"`
 	// Syntax this transform was written in
+	// Deprecated - syntax is defined per-step
 	Syntax string `json:"syntax,omitempty"`
 	// SyntaxVersion is an identifier for the application and version number that
 	// produced the result
-	SyntaxVersion string `json:"syntaxVersion,omitempty"`
+	// Deprecated - use steps.Syntax with a version suffix instead
+	SyntaxVersion map[string]string `json:"syntaxVersion,omitempty"`
```

Comment:
As written this would be a breaking change. We need to _keep_ `SyntaxVersion` and _add_ a field. While we talked about naming this `SyntaxVersions`, I think using `Syntaxes` as a field name would be nicer: 

```go
type Transform struct {
  // ...
  // map of syntaxes used in this transform to their version identifier.
  Syntaxes map[string]string `json:"syntaxes,omitempty"`

Reply:
oh my bad i missed this naming distinction. I did do it in a separate commit because it felt very dramatic and wanted to be able to walk it back easily.